src/Uno.Toolkit.UI/Behaviors/ItemsRepeaterExtensions.IncrementalLoading.cs

```diff
@@ -0,0 +1,231 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Uno.Disposables;
+using Uno.Extensions;
+using Uno.Logging;
+
+#if IS_WINUI
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI.Xaml.Controls.Primitives;
+using Microsoft.UI.Xaml.Input;
+
+#else
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using ItemsRepeater = Microsoft.UI.Xaml.Controls.ItemsRepeater;
+using UniformGridLayout = Microsoft.UI.Xaml.Controls.UniformGridLayout;
+using StackLayout = Microsoft.UI.Xaml.Controls.StackLayout;
+
+#if HAS_UNO
+using FlowLayout = Microsoft.UI.Xaml.Controls.FlowLayout;
+#endif
+
+#endif
+
+namespace Uno.Toolkit.UI
+{
+	// Partial class to separate the IncrementalLoading behavior from the main ItemsRepeaterExtensions class
+	partial class ItemsRepeaterExtensions
+	{
+		#region DependencyProperty: SupportsIncrementalLoading
+
+		/// <summary>
+		/// Gets or sets whether the ItemsRepeaterExtensions should automatically prefetch more items when the user scrolls to the end of the list.
+		/// </summary>
+		public static DependencyProperty SupportsIncrementalLoadingProperty { [DynamicDependency(nameof(GetSupportsIncrementalLoading))] get; } = DependencyProperty.RegisterAttached(
+			"SupportsIncrementalLoading",
+			typeof(bool),
+			typeof(ItemsRepeaterExtensions),
+			new PropertyMetadata(default(bool), OnSupportsIncrementalLoadingChanged));
+
+		[DynamicDependency(nameof(SetSupportsIncrementalLoading))]
+		public static bool GetSupportsIncrementalLoading(ItemsRepeater obj) => (bool)obj.GetValue(SupportsIncrementalLoadingProperty);
+		[DynamicDependency(nameof(GetSupportsIncrementalLoading))]
+		public static void SetSupportsIncrementalLoading(ItemsRepeater obj, bool value) => obj.SetValue(SupportsIncrementalLoadingProperty, value);
+
+		#endregion
+
+		#region DependencyProperty: IsLoading
+
+		/// <summary>
+		/// Gets whether there is currently a prefetch operation in progress.
+		/// </summary>
+		public static DependencyProperty IsLoadingProperty { [DynamicDependency(nameof(GetIsLoading))] get; } = DependencyProperty.RegisterAttached(
+			"IsLoading",
+			typeof(bool),
+			typeof(ItemsRepeaterExtensions),
+			new PropertyMetadata(default(bool)));
+
+		[DynamicDependency(nameof(SetIsLoading))]
+		public static bool GetIsLoading(ItemsRepeater obj) => (bool)obj.GetValue(IsLoadingProperty);
+		[DynamicDependency(nameof(GetIsLoading))]
+		private static void SetIsLoading(ItemsRepeater obj, bool value) => obj.SetValue(IsLoadingProperty, value);
```

Comment:
iirc, there is a bug when the g/setter doesnt have the same signature for dp
```suggestion
		public static void SetIsLoading(ItemsRepeater obj, bool value) => obj.SetValue(IsLoadingProperty, value);
```